packages/react-cookie/src/hooks.ts

```diff
@@ -0,0 +1,56 @@
+import {useContext, useState} from 'react';
+import {CookieSerializeOptions} from 'cookie';
+import {CookieContext} from './context';
+
+const NO_MANAGER_ERROR = [
+  'Could not find react-cookie context.',
+  'You might be missing the <Cookie />',
+  'component in your application component tree.',
+].join(' ');
+
+export function useCookie(
+  key: string,
+): [
+  string | undefined,
+  (value?: string, options?: CookieSerializeOptions) => void
+] {
+  const manager = useContext(CookieContext);
+
+  if (!manager) {
+    throw new Error(NO_MANAGER_ERROR);
+  }
+
+  const [cookie, setCookieValue] = useState(
+    () => (manager ? manager.getCookie(key) : undefined),
+  );
+
+  const setCookie = (value?: string, options?: CookieSerializeOptions) => {
+    if (!manager) {
```

Comment:
Manager should always be defined given that we are throwing before we construct this when it isn't.

This is actually a great usecase for the upcoming [assert signatures](https://httptoolkit.tech/blog/5-big-features-of-typescript-3.7/) feature, but till we have that I'd just `manager!` or cast it.

Reply:
on a side note it feels kinda weird that this lives here and not in the manager itself.